packages/extension-onboarding/src/utils/AsyncQue.ts

```diff
@@ -0,0 +1,40 @@
+import { ResultAsync } from "neverthrow";
+
+interface QueueItem<T, E> {
+  asyncFunction: ResultAsync<T, E>;
+  resolve: (value: void | PromiseLike<void>) => void;
+  reject: (reason?: never) => void;
+}
+
+export class AsyncQueue {
```

Comment:
Should this go into, maybe, Common Utils? Seems like it might be something we can use outside of extension-onboarding.